graphql2/graphqlapp/messagelog.go

```diff
@@ -113,18 +118,35 @@ func (q *MessageLogConnectionStats) TimeSeries(ctx context.Context, opts *notifi
 		origin = *input.BucketOrigin
 	}
 
+	var s string
+	if segmentBy != nil {
+		value, err := segmentBy.Value()
+		if err != nil {
+			return nil, err
+		}
+		s = value.(string)
+	}
 	buckets, err := q.NotificationStore.TimeSeries(ctx, notification.TimeSeriesOpts{
 		SearchOptions:      *opts,
 		TimeSeriesInterval: dur,
 		TimeSeriesOrigin:   origin,
+		SegmentBy:          s,
 	})
 	if err != nil {
 		return nil, err
 	}
 
 	out := make([]graphql2.TimeSeriesBucket, len(buckets))
 	for i, b := range buckets {
-		out[i] = graphql2.TimeSeriesBucket(b)
+
+		var gb graphql2.TimeSeriesBucket
+
+		gb.Count = b.Count
+		gb.Start = b.Start
+		gb.End = b.End
+		gb.SegmentLabel = b.SegmentLabel
+
+		out[i] = graphql2.TimeSeriesBucket(gb)
```

Comment:
Can this be simplified to `out[i] = graphql2.TimeSeriesBucket(b)`? They should have the same fields